### PR TITLE
Make node release binaries debuggable

### PR DESCRIPTION
**Pull request by bsudekum**
This change makes Node release binaries a little bit more debuggable. Now when there is a crash in production, we should get more information including line numbers. There is no speed implications here, the only downside is an increase in binary size (currently, release binaries are ~2mb).

/cc @tmpsantos @brunoabinader

### CODE REVIEW

**Comment by bsudekum** [comment]
Heh, well we got a [coredump during tests](https://travis-ci.org/mapbox/mapbox-gl-native/jobs/237978675#L1323), includes a little more information:

```
Thread 1 (Thread 0x2b30157301c0 (LWP 11886)):
#0  v8::Object::GetInternalField (this=0x0, index=1) at node/v6.10.3/include/v8.h:8017
#1  node_mbgl::NodeRequest::doExecute (this=0x598d770) at ../../../platform/node/src/node_request.cpp:136
#2  0x00002b3028348d9c in node_mbgl::NodeRequest::Execute()::$_0::operator()() const (this=<optimized out>) at ../../../platform/node/src/node_request.cpp:125
#3  std::_Function_handler<void (), node_mbgl::NodeRequest::Execute()::$_0>::_M_invoke(std::_Any_data const&) (__functor=...) at /usr/bin/../lib/gcc/x86_64-linux-gnu/5.4.1/../../../../include/c++/5.4.1/functional:1871
#4  0x000000000131cc1b in uv__async_event (loop=0x1e1efc0 <default_loop_struct>, w=<optimized out>, nevents=<optimized out>) at ../deps/uv/src/unix/async.c:98
#5  0x000000000131ccf3 in uv__async_io (loop=0x1e1efc0 <default_loop_struct>, w=0x1e1f188 <default_loop_struct+456>, events=<optimized out>) at ../deps/uv/src/unix/async.c:138
#6  0x000000000132d2c0 in uv__io_poll (loop=0x1e1efc0 <default_loop_struct>, timeout=0) at ../deps/uv/src/unix/linux-core.c:380
#7  0x000000000131d7d6 in uv_run (loop=0x1e1efc0 <default_loop_struct>, mode=UV_RUN_ONCE) at ../deps/uv/src/unix/core.c:354
#8  0x00000000010a62d0 in node::Start(int, char**) ()
#9  0x00002b3016d6ff45 in __libc_start_main () from /lib/x86_64-linux-gnu/libc.so.6
#10 0x00000000007b8b0d in _start ()
```

File: platform/node/scripts/after_success.sh [hunk 1]
```diff
@@ -4,7 +4,7 @@ set -e
 set -o pipefail
 
 if [[ -n ${PUBLISH:-} ]]; then
-    if [[ "${BUILDTYPE}" == "Release" ]]; then
+    if [[ "${BUILDTYPE}" == "RelWithDebInfo" ]]; then
```
Comment (springmeyer) [inline platform/node/scripts/after_success.sh]: To keep this working with `Release` how about:

```
if [[ "${BUILDTYPE}" == "Release" ]] || [[ "${BUILDTYPE}" == "RelWithDebInfo" ]]; then
```

Comment (bsudekum) [inline platform/node/scripts/after_success.sh]: @springmeyer yeah I thought about this, but I'm sure if the code path `if [[ "${BUILDTYPE}" == "Release" ]]` would ever be hit. With this PR, we'd be removing `Release` completely.

Comment (springmeyer) [inline platform/node/scripts/after_success.sh]: True. What I am concerned about is the fallback to Debug. How about this instead:

```
if [[ -n ${PUBLISH:-} ]]; then
    if [[ "${BUILDTYPE}" == "RelWithDebInfo" ]]; then
        ./node_modules/.bin/node-pre-gyp package publish info
    elif [[ "${BUILDTYPE}" == "Debug" ]]; then
        ./node_modules/.bin/node-pre-gyp package publish info --debug
    else
        echo "error: must provide either Debug or RelWithDebInfo for BUILDTYPE"
        exit 1
    fi
fi
```

Comment (brunoabinader) [inline platform/node/scripts/after_success.sh]: @springmeyer @bsudekum is this logic also feasible to go into https://github.com/mapbox/mapbox-gl-native/pull/9051 ?

**Comment by springmeyer** [comment]
I also recommend adding:

```
SET(CMAKE_CXX_FLAGS_RELWITHDEBINFO "-O3 -DNDEBUG")
```

right after https://github.com/mapbox/mapbox-gl-native/blob/893584517182c270b7238a2fddb616a8a0e657d6/CMakeLists.txt#L75

The reason is that I think with `RelWithDebInfo` the variables from `CMAKE_CXX_FLAGS_RELEASE` will no longer be used. And the `-Os` asks the compiler to build binaries with the approach of `When in doubt, prefer small, not fast code.`. This was added in https://github.com/mapbox/mapbox-gl-native/commit/8742dc852c82ef40003a22a8e0a764616d8beed1 to reduce binary size with mobile in mind. But because binary size does not matter for node-mbgl we should be building with `-O3` /cc @jfirebaugh

**Comment by brunoabinader** [comment]
Follow-up in #9497.

**Pull request closed without merging**